Update release-trueos.sh

beanpole135 opened the pull request:
Add a "super_clean_poudriere" step after "clean_poudriere".
This detects/fixes an issue that has appeared on the Trident build server a few times, where poudriere does not clean up the jail "all-the-way", and leaves a dangling dataset/mountpoint/directory without the corresponding configuration files. This causes the jail creation routine in the next step to fail because the mountpoint from the previous job still exists.

beanpole135 committed:
Update release-trueos.sh

Add a "super_clean_poudriere" step after "clean_poudriere".
This detects/fixes an issue that has appeared on the Trident build server a few times, where poudriere does not clean up the jail "all-the-way", and leaves a dangling dataset/mountpoint/directory without the corresponding configuration files. This causes the jail creation routine in the next step to fail because the mountpoint from the previous job still exists.

kmoore134 commented on release/release-trueos.sh:
```diff
@@ -161,6 +161,23 @@ clean_poudriere()
+	#Look for any leftover mountpoints/directories and remove them too
+	for i in `ls ${POUDRIERED_DIR}/*-make.conf`
+	do
+		name=`basename "${i}" | rev | cut -d "-" -f 2-10 | rev`
```
This looks *real* dangerous. If there are more or less '-' characters then something could go wrong here. 

Recommend you cd ${POUDRIERED_DIR} and ls '*-make.conf' instead. Then to get the jail/base name, you can do echo $i | sed 's|-make.conf||g' without that horrible rev / cut / rev stuff ;)

kmoore134 replied:
Nevermind on the "cd" part, didn't see the 'basename' section.

beanpole135 replied:
Done!

kmoore134 commented on release/release-trueos.sh:
```diff
@@ -161,6 +161,23 @@ clean_poudriere()
+			_stale_dataset=$(mount | grep 'on ${POUDRIERE_BASEFS}/jails/${name} ' | cut -w -f 1)
+			umount ${POUDRIERE_BASEFS}/jails/${name}
+			rmdir ${POUDRIERE_BASEFS}/jails/${name}
+			zfs destroy -r ${_stale_dataset}
```
Need some error checking here. You probably want to validate _stale_dataset exists or isn't bogus before you start running zfs destroy.

beanpole135 replied:
done!

beanpole135 committed:
Update release-trueos.sh

Cleanup a bit of the basename cleanup, and add some extra checks to the ZFS dataset validation/removal.
